sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -488,6 +488,12 @@ private[sql] case class EnsureRequirements(sqlContext: SQLContext) extends Rule[
   }
 
   def apply(plan: SparkPlan): SparkPlan = plan.transformUp {
+    case operator @ Exchange(partitioning, child, _) =>
+      child.children match {
+        case Exchange(childPartitioning, baseChild, _)::Nil =>
```

Comment:
Hey, so I know that I said to add the logic here, but now I'm realizing its kind of odd to add it into the plan and then remove it.  Why don't we just never add it?  Maybe this logic should go in SparkStrategies?